src/server/models/EventVolunteer.ts

```diff
@@ -0,0 +1,26 @@
+import { EventVolunteer } from '@/types/dataModel/eventVolunteer';
+import { Model, Schema, model, models } from 'mongoose';
+
+const EventVolunteerSchema = new Schema({
+  volunteer: {
+    type: Schema.Types.ObjectId,
+    ref: 'Volunteer',
+    required: true,
+  },
+  event: {
+    type: Schema.Types.ObjectId,
+    ref: 'Event',
+    required: true,
+  },
+  checkInTime: {
+    type: Date,
+  },
+  checkOutTime: {
+    type: Date,
+  },
+});
+
+export type EventVolunteerDocument = EventVolunteer & Document;
+
+export default (models.EventHours as Model<EventVolunteerDocument>) ||
```

Comment:
this looks good, but you need to change models.EventHours to models.EventVolunteer